tests/lib/prepare.sh

```diff
@@ -0,0 +1,147 @@
+#!/bin/bash
+
+set -eux
+
+prepare_classic() {
+    apt install -y ${SPREAD_PATH}/../snapd_*.deb
+    # Snapshot the state including core.
+    if [ ! -f $SPREAD_PATH/snapd-state.tar.gz ]; then
+        ! snap list | grep core || exit 1
```

Comment:
This is to ensure we do not snapshot a state that is incomplete (i.e. core is missing). The purpose of the state caching is to avoid the 60mb core download all the time :)